client/engine/chainservice/mock_chain.go

```diff
@@ -0,0 +1,74 @@
+package chainservice
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/statechannels/go-nitro/client/engine/store/safesync"
+	"github.com/statechannels/go-nitro/protocols"
+	"github.com/statechannels/go-nitro/types"
+)
+
+// MockChain is an in-memory "chain" that accepts transactions and broadcasts events
+type MockChain interface {
```

Comment:
nit: is this interface necessary? It seems like there'd only ever be one implementation of it -- would it be cleaner to delete this interface and rename `MockChainImp` to `MockChain`?

Reply:
Good call, done in 37d7c771.